jubilant/_juju.py

```diff
@@ -678,9 +679,15 @@ def wait(
 
         while time.monotonic() - start < timeout:
             prev_status = status
-            status = self.status()
+
+            stdout, _ = self._cli('status', '--format', 'json', log=False)
```

Comment:
At first I thought this was a potential code smell: is self.status() is not longer used, why not remove it too?

I guess the thinking here is that public user-facing Juju.status() is detached from its use in this wait function.

I guess it's fine. I hope the use here (cli/json/from_dict) doesn't diverge from that in `status()` over time.

In any case, this approach is better than propagating `log=False` to "arbitrary" functions like `.status()`. Python-libjuju suffers from a case of that.

👍🏻

Reply:
Yeah, the idea is that `Juju.status` is exposed separately. I could create an `_status` helper that both call, but it didn't feel worth it to save two lines.